core/model/modx/processors/resource/sort.class.php

```diff
@@ -39,6 +40,8 @@ public function process() {
         $this->getNodesFormatted($data, $this->getProperty('parent', 0));
 
         $this->fireBeforeSort();
+        $resourceBeforeSort = $this->modx->getObject('modResource', array('id' => $this->getProperty('source_pk')));
```

Comment:
I believe this processor is also handling context sorting, so `source_pk` could be a `modContext.key`